pkg/registry/deployment/etcd/etcd_test.go

```diff
@@ -180,73 +184,69 @@ func TestWatch(t *testing.T) {
-// 			Replicas: validDeployment.Status.Replicas,
-// 			Selector: validDeployment.Spec.Template.Labels,
-// 		},
-// 	}
-// }
-
-// var validScale = *validNewScale()
-
-// func TestScaleGet(t *testing.T) {
-// 	storage, server := newStorage(t)
-// 	defer server.Terminate(t)
-
-// 	ctx := api.WithNamespace(api.NewContext(), namespace)
-// 	key := etcdtest.AddPrefix("/deployments/" + namespace + "/" + name)
-// 	if err := storage.Deployment.Storage.Set(ctx, key, &validDeployment, nil, 0); err != nil {
-// 		t.Fatalf("unexpected error: %v", err)
-// 	}
-
-// 	expect := &validScale
-// 	obj, err := storage.Scale.Get(ctx, name)
-// 	if err != nil {
-// 		t.Fatalf("unexpected error: %v", err)
-// 	}
-// 	scale := obj.(*extensions.Scale)
-// 	if e, a := expect, scale; !api.Semantic.DeepDerivative(e, a) {
-// 		t.Errorf("unexpected scale: %s", util.ObjectDiff(e, a))
-// 	}
-// }
-
-// func TestScaleUpdate(t *testing.T) {
-// 	storage, server := newStorage(t)
-// 	defer server.Terminate(t)
-
-// 	ctx := api.WithNamespace(api.NewContext(), namespace)
-// 	key := etcdtest.AddPrefix("/deployments/" + namespace + "/" + name)
-// 	if err := storage.Deployment.Storage.Set(ctx, key, &validDeployment, nil, 0); err != nil {
-// 		t.Fatalf("unexpected error: %v", err)
-// 	}
-// 	replicas := 12
-// 	update := extensions.Scale{
-// 		ObjectMeta: api.ObjectMeta{Name: name, Namespace: namespace},
-// 		Spec: extensions.ScaleSpec{
-// 			Replicas: replicas,
-// 		},
-// 	}
-
-// 	if _, _, err := storage.Scale.Update(ctx, &update); err != nil {
-// 		t.Fatalf("unexpected error: %v", err)
-// 	}
-// 	obj, err := storage.Deployment.Get(ctx, name)
-// 	if err != nil {
-// 		t.Fatalf("unexpected error: %v", err)
-// 	}
-// 	deployment := obj.(*extensions.Deployment)
-// 	if deployment.Spec.Replicas != replicas {
-// 		t.Errorf("wrong replicas count expected: %d got: %d", replicas, deployment.Spec.Replicas)
-// 	}
-// }
+func TestScaleGet(t *testing.T) {
+	storage, server := newStorage(t)
+	defer server.Terminate(t)
+
+	ctx := api.WithNamespace(api.NewContext(), namespace)
+	key := etcdtest.AddPrefix("/deployments/" + namespace + "/" + name)
+	if err := storage.Deployment.Storage.Set(ctx, key, &validDeployment, nil, 0); err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
+
+	selector, err := unversioned.LabelSelectorAsSelector(validDeployment.Spec.Selector)
+	if err != nil {
+		t.Errorf("Invalid deployment selector %+v: %v", validDeployment.Spec.Selector, err)
+	}
+	want := &autoscaling.Scale{
+		ObjectMeta: api.ObjectMeta{Name: name, Namespace: namespace},
+		Spec: autoscaling.ScaleSpec{
+			Replicas: validDeployment.Spec.Replicas,
+		},
+		Status: autoscaling.ScaleStatus{
+			Replicas: validDeployment.Status.Replicas,
+			Selector: selector.String(),
+		},
+	}
+	obj, err := storage.Scale.Get(ctx, name)
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
+	got := obj.(*autoscaling.Scale)
+	if !api.Semantic.DeepDerivative(want, got) {
```

Comment:
I think I want DeepDerivative but have switched to DeepEqual by filling in the fields which were nil/empty in `want` previously. Just to be safe.